src/modules/ekf2/EKF/covariance.cpp

```diff
@@ -163,22 +163,26 @@ void Ekf::predictCovariance(const imuSample &imu_delayed)
 
 
 #if defined(CONFIG_EKF2_MAGNETOMETER)
-	if (_control_status.flags.mag) {
-		// mag_I: add process noise
-		float mag_I_sig = dt * math::constrain(_params.mage_p_noise, 0.f, 1.f);
-		float mag_I_process_noise = sq(mag_I_sig);
+	// mag_I: add process noise
```

Comment:
My point of view on this is that I don't see why we would keep the uncertainty low on a state we don't observe. I agree that we should have an upper limit, but I would do the same for the IMU bias and wind: it's not because we can't measure the wind that we should be sure about its velocity for a long time.